src/App.tsx

```diff
@@ -1,56 +1,108 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Person = {
+  name: string,
+  sex: string,
+  born: number,
+  died: number,
+  fatherName: string | null,
+  motherName: string | null,
+  slug: string,
+};
+
+type Props = {
+  delay: number,
+};
+
+export const App: React.FC<Props> = ({ delay }) => {
+  const [query, setQuery] = useState('');
+
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const [isFocused, setIsFocused] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handleButtonClick = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer
+      .filter(personFromServer => personFromServer
+        .name.toLowerCase().includes(appliedQuery.toLowerCase()));
+  }, [appliedQuery]);
 
   return (
     <main className="section">
-      <h1 className="title">
-        {`${name} (${born} = ${died})`}
-      </h1>
-
-      <div className="dropdown is-active">
+      {selectedPerson && query === selectedPerson.name ? (
+        <h1 className="title">
+          {`${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`}
+        </h1>
+      ) : (
+        <h1 className="title">
+          No selected person
+        </h1>
+      )}
```

Comment:
Also, in order not to duplicate the same tag with the same class twice, it would be better to move this check inside the tag